### PR TITLE
feat: Add the Hamburger Menu for the Portfolio mobile version

### PR DESCRIPTION
**Pull request by gvgesanto2**
## Description

**Task:** https://github.com/microverseinc/curriculum-html-css/blob/main/portfolio/7_mobile_menu.md

This PR makes the following changes:
- Translates all of the CSS code to Sass
- Adds a Hamburger Menu for the mobile version of the Portfolio homepage
- Adds the functionality of opening and closing the Hamburger Menu
-  Sets up the GitHub Actions and the [ESLint](https://eslint.org/) (to lint the JS files) and [Stylelint](https://stylelint.io/) (to just lint the SCSS files) linters

Plus, it adopts the following methodologies and conventions:
- [BEM](https://en.bem.info/methodology/) methodology for naming the CSS classes
- A modified version of the [Scalable and Modular Architecture for CSS](http://smacss.com/) (SMACSS) methodology for organizing the CSS code into categorized folders/files (base, components, layout, and utilities)
- [Atomic Design](https://bradfrost.com/blog/post/atomic-web-design/) methodology for breaking down the CSS components into atoms, molecules, and organisms

**Note: Please be completely honest about the quality of my code. Would you do something different? Is there a better way to solve this problem that you know about? Please let me know! I love receiving feedback with tips, suggestions, better solutions, or better ways to approach the problem (even when I implemented all the project requirements), because I believe it makes me a better developer.**

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have commented my code
- [X] I have deleted any unused code as inline comments in my codebase

## Screenshots

hambMenuScreenshot

### CODE REVIEW

File: scss/abstracts/_variables.scss [hunk 1]
```diff
@@ -51,3 +52,34 @@
...
+  'screen-xlg': 68.75em,
+  // 75em = 1200px
+  'screen-xxlg': 75em,
+  // 81.25em = 1300px
+  'screen-hg': 81.25em,
+  // 93.75em = 1500px
+  'screen-xhg': 93.75em,
+  // 100em = 1600px
+  'screen-xxhg': 100em
+);
```
Comment (zieeco) [inline scss/abstracts/_variables.scss:80]: - [ ] **[REQUIRED CHANGES]**

- Please, kindly remove unused code, it makes your codebase not clean.

File: scss/base/_base.scss [hunk 2]
```diff
@@ -1,34 +1,21 @@
...
-  /* 1rem = 10px */
+  // Setting the value of 1rem to 10px
   font-size: 62.5%;
-}
+  box-sizing: border-box;
 
-/* 20em = 320px */
-@media only screen and (max-width: 20em) {
-  html {
+  // screen-xsm = 20em = 320px
```
Comment (zieeco) [inline scss/base/_base.scss:6]: - [ ] **[REQUIRED CHANGES]**

- Please, kindly remove unused code, it makes your codebase not clean.

File: scss/base/_base.scss [hunk 3]
```diff
@@ -1,34 +1,21 @@
...
+  @include respond-to('max', 'screen-xsm') {
     font-size: 52.25%;
   }
-}
 
-/* 15.625em = 250px */
-@media only screen and (max-width: 15.625em) {
-  html {
+  // screen-xxsm = 15.625em = 250px
```
Comment (zieeco) [inline scss/base/_base.scss:11]: - [ ] **[REQUIRED CHANGES]**

- Please, kindly remove unused code, it makes your codebase not clean.

File: scss/base/_base.scss [hunk 4]
```diff
@@ -1,34 +1,21 @@
...
-*::before,
-*::after {
-  box-sizing: border-box;
-  margin: 0;
-  padding: 0;
-}
-
-/* Setting the value of 1rem */
 html {
-  /* 1rem = 10px */
```
Comment (zieeco) [inline scss/base/_base.scss:12]: - [ ] **[REQUIRED CHANGES]**

- Please, kindly remove unused code, it makes your codebase not clean.

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team


Great work on your mobile menu 👏 
You've done well implementing it, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice design. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gvgesanto2** [comment]
## Hi @zieeco,

Thank you for your review and feedback! But, I got a question about the changes you demanded... You are asking me to remove unused code from my codebase, however, **the lines of code you referenced in your review are NOT "unused code"**, they're just comments saying how much some value of  ``em`` units is in pixels (``px``). 

**I didn't leave any "unused code" in my codebase, so please, it would be very nice if you could re-evaluate your review.**

**Review by sinansevgi — COMMENTED** [review]
Hello @gvgesanto2  👋,

_This is an additional review as a result of the second opinion request._

I can see that your comments are **not** `commented-out code blocks`. You use comments to give references about sizing, which is ok to have in your code.  
I still suggest you keep your comments concise to make your code more readable. But since these comments are reference comments, I think your project should get approved.

As this was the only reason that your project was not approved it will be approved as soon as possible.
Please note that since we will approve it manually so this could take a few hours!

Keep rocking 🚀!

**Pull request merged**